input.go

```diff
@@ -344,6 +363,66 @@ func (w *Window) SetCursorPosition(xpos, ypos float64) error {
 	return fetchError()
 }
 
+// Creates a new custom cursor image that can be set for a window with SetCursor.
+// The cursor can be destroyed with Destroy. Any remaining cursors are destroyed by Terminate.
+//
+// The pixels are 32-bit little-endian RGBA, i.e. eight bits per channel. They are arranged
+// canonically as packed sequential rows, starting from the top-left corner.
+// 
+// All non-RGBA images will be converted to RGBA.
+//
+// The cursor hotspot is specified in pixels, relative to the upper-left corner of the cursor image.
+// Like all other coordinate systems in GLFW, the X-axis points to the right and the Y-axis points down.
+func CreateCursor(img image.Image, xhot, yhot int) (*Cursor, error) {
```

Comment:
byte string is not enough. you also need the width and height as in
GLFWimage struct. We can just create an Image type analog to GLFWimage and
be done with it. We can leave creating it to the developer or we can
include a helper CreateImage function that takes byte data width and height
as input.
On 20 Jan 2015 07:25, "Dmitri Shuralyov" <notifications@github.com> wrote:

> In input.go
> <https://github.com/go-gl/glfw3/pull/116#discussion-diff-23203525>:
>
> > @@ -344,6 +363,66 @@ func (w *Window) SetCursorPosition(xpos, ypos float64) error {
> >  	return fetchError()
> >  }
> >
> > +// Creates a new custom cursor image that can be set for a window with SetCursor.
> > +// The cursor can be destroyed with Destroy. Any remaining cursors are destroyed by Terminate.
> > +//
> > +// The pixels are 32-bit little-endian RGBA, i.e. eight bits per channel. They are arranged
> > +// canonically as packed sequential rows, starting from the top-left corner.
> > +//
> > +// All non-RGBA images will be converted to RGBA.
> > +//
> > +// The cursor hotspot is specified in pixels, relative to the upper-left corner of the cursor image.
> > +// Like all other coordinate systems in GLFW, the X-axis points to the right and the Y-axis points down.
> > +func CreateCursor(img image.Image, xhot, yhot int) (*Cursor, error) {
>
> I'm still thinking whether it makes a better API/experience to accept
> image.Image instead of just imageData []byte...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/go-gl/glfw3/pull/116/files#r23203525>.
>